keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -96,16 +111,35 @@ data class ResourceUpdated(
   val delta: Map<String, Any?>,
   override val timestamp: Instant
 ) : ResourceEvent() {
-  constructor(resource: Resource<*>, delta: Map<String, Any?>, clock: Clock) : this(
+  constructor(resource: Resource<*>, delta: Map<String, Any?>, clock: Clock = Companion.clock) : this(
     resource.uid,
     resource.apiVersion,
     resource.kind,
     resource.name.value,
     delta,
     clock.instant()
   )
+}
+
+data class ResourceDeleted(
+  override val uid: UID,
+  override val apiVersion: ApiVersion,
+  override val kind: String,
+  override val name: String,
+  override val timestamp: Instant
+) : ResourceEvent() {
+  constructor(resource: Resource<*>, clock: Clock = Companion.clock) : this(
+    resource.uid,
+    resource.apiVersion,
+    resource.kind,
+    resource.name.value,
+    clock.instant()
+  )
+}
 
-  constructor(resource: Resource<*>, delta: Map<String, Any?>) : this(resource, delta, clock)
+abstract class ResourceCheckResult : ResourceEvent() {
+  abstract val state: ResourceState
+  override val ignoreRepeatedInHistory = true
```

Comment:
Added this intermediate abstract class (which is the thing Atlas telemetry is interested in).

Reply:
Also we don't want > 1 of these in a row in the events table.